### DISCUSSION TITLE
Designing a smartphone in Akira

### DISCUSSION THREAD

**Discussion by AUNaseef** [akiraux/Akira#472] [General]
As requested by @albfan on Twitter, I decided to design something with Akira. Since the Librem 5 just started shipping, I thought I'd design a phone. So here is a smartphone designed in Akira. With the mandatory Tux on the right. 
![Screenshot from OpenPhone](https://user-images.githubusercontent.com/17310513/66671845-9024b200-ec7a-11e9-99c8-f6a888d9204d.png)
[Phone.zip](https://github.com/akiraux/Akira/files/3718877/OpenPhoneUpdated.zip)

I started the design with a blue rectangle for the screen, with a rotated rectangle at the top, and some rectangles the same colors as the background to hide the parts that aren't necessary.

The red "A" in the middle, which should not have been red, was designed the same way, just a few rectangles making the basic shape and background-colored rectangles hiding the parts that aren't necessary.

The round-corned shapes are 2 rectangles placed on top of each other with ellipses (almost circles) at all 4 corners. It was kinda hard to get the sizes right, I'd like to be able to hold *Ctrl* or *Shift* key to make a perfect circle and maybe input the radius manually or make duplicates of the existing objects so sizes will be consistent.

Ability to snap edges to circle's center points could've saved me so much time. I would also love to see undo/redo feature implemented in Akira soon, it can be really useful most of the time.

Most, if not all of the things I say are probably things the developers are already working on, but I'm going to list them all here any way.

I am not a UI/UX designer, I use Akira to draw things, making art. even tho Akira is faster to work with and it is much easier to design something that looks good in it, it doesn't have all the features I need right now. So it would be nice if I could export my work as an .svg and add some final touches to them in Inkscape, this feature could also make the things designed in Akira actually usable.

Another thing that could've saved me so much time (other than a grid or snapping positions) is the ability to snap the rotation angles to nice round increments while holding the *ctrl* key. And also, I want to be able to move the selected shape by ~16px using arrow keys, and ~1-2px when holding *shift*

Designing the Tux was really straightforward, it is mostly just ellipses with just one rectangle for the bottom part of the heart.

**Comment by AUNaseef** [comment]
I had a few problems while designing the phone:
Zooming in and out with *ctrl* + *mouse wheel* (AkiraUX tweeted about this so I think it won't be so long until it's fixed)
Scrolling left-right while holding the *shift* key
Panning the canvas by holding the *middle mouse button*
Modifying the content of Text and deleting the text when I don't need it (Workaroud : hiding it from the artboard)
The artboard isn't populated when opening a saved file, and deleting an object from the artboard doesn't remove it from the canvas.
[URGENT] Changing color, border size, border color and Z position (idk what you call it) after creating the object.
Oh and transparency.

**Comment by albfan** [comment]
Thanks @AUNaseef. This feedback is really valuable. Most of the things you need are already implemented in opened PRs.

We will update this use case when things get landed on master.

Screenshot looks amazing!

**Comment by b-fuze** [comment]
So I finally got Akira compiled and running on Ubuntu 18.04 LTS, great work btw:
![image](https://user-images.githubusercontent.com/10504957/67068155-20fe0080-f147-11e9-97e8-e1634a368655.png)

But I'm not sure how to add color to shapes? Or is that in a separate branch?

**Comment by albfan** [comment]
@b-fuze yes, it is a separate branch
